identity-service/src/routes/tiktok.js

```diff
@@ -57,9 +64,159 @@ module.exports = function (app) {
+              uuid: tikTokUser.open_id,
+              profile: tikTokUser,
+              verified: tikTokUser.is_verified
+            })
+
+            return successResponse(accessTokenResponse.data)
+          } catch (err) {
+            return errorResponseBadRequest(err)
+          }
+        }
+      } catch (err) {
+        return errorResponseBadRequest(err)
+      }
+    })
+  )
 
-      return successResponse(resp.data)
+  /**
+   * After the user finishes onboarding in the client app and has a blockchain userId, we need to associate
+   * the blockchainUserId with the tiktok profile so we can write the verified flag on chain
+   */
+  app.post(
+    '/tiktok/associate',
+    handleResponse(async (req, res, next) => {
+      const { uuid, userId, handle } = req.body
+      const audiusLibsInstance = req.app.get('audiusLibs')
+
+      try {
+        const tikTokObj = await models.TikTokUser.findOne({
+          where: { uuid: uuid }
+        })
+
+        const user = await models.User.findOne({
+          where: { handle }
+        })
+
+        const isUnassociated = tikTokObj && !tikTokObj.blockchainUserId
+        const handlesMatch =
+          tikTokObj &&
+          tikTokObj.profile.display_name.toLowerCase() ===
+            user.handle.toLowerCase()
+
+        // only set blockchainUserId if not already set
+        if (isUnassociated && handlesMatch) {
+          tikTokObj.blockchainUserId = userId
+
+          // if the user is verified, write to chain, otherwise skip to next step
+          if (tikTokObj.verified) {
+            const [encodedABI, contractAddress] =
+              await audiusLibsInstance.User.updateIsVerified(
+                userId,
+                true,
+                config.get('userVerifierPrivateKey'),
+                config.get('entityManagerReplicaSetEnabled')
+              )
+            const contractRegKey =
+              await audiusLibsInstance.contracts.getRegistryContractForAddress(
+                contractAddress
+              )
+            const senderAddress = config.get('userVerifierPublicKey')
+
+            try {
+              const txProps = {
+                contractRegistryKey: contractRegKey,
+                contractAddress: contractAddress,
+                encodedABI: encodedABI,
+                senderAddress: senderAddress,
+                gasLimit: null
+              }
+              await txRelay.sendTransaction(
+                req,
+                false,
+                txProps,
+                'tikTokVerified'
+              )
+              await verifiedUserReporter.report({ userId, handle })
+            } catch (e) {
+              return errorResponseBadRequest(e)
+            }
+          }
+
+          const socialHandle = await models.SocialHandles.findOne({
+            where: { handle }
+          })
+          if (socialHandle) {
+            socialHandle.tikTokHandle = tikTokObj.profile.display_name
```

Comment:
No toLowerCase for this? I guess since it's already lowercase?

Reply:
I think here we want to keep the casing so it shows up properly on the users profile page